pylint/config/options_provider_mixin.py

```diff
@@ -72,10 +73,12 @@ def set_option(self, optname, value, action=None, optdict=None):
                 setattr(self.config, optname, _list + (value,))
             else:
                 _list.append(value)
-        elif action == "callback":
-            optdict["callback"](None, optname, value, None)
-        elif not isinstance(action, str):
-            optdict["callback"](None, optname, value, None)
+        elif (
```

Comment:
Similar as above. This makes it so that `optparse` no longer fires prematurely.

I don't think these options should fire anyway. If you using `set_option` to call a callback function you're doing something wrong anyway.